src/Robot.Drivers/Sonar/HcSr04DistanceSensorDriver.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Iot.Device.Hcsr04;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Device.Gpio;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Robot.Drivers.Sonar
+{
+    public class HcSr04DistanceSensorDriver : IDistanceSensor
+    {
+        private Hcsr04 Sensor;
+
+        public int Id { get; }
+        public double Angle { get; }
+        public CancellationTokenSource MeasuringCancellationToken { get; private set; }
+        public HcSr04DistanceSensorDriverSettings Settings { get; }
+        public ILogger<HcSr04DistanceSensorDriver> Logger { get; }
+
+        public event EventHandler<SonarDistanceEventArgs> SonarDistanceChanged;
+
+        public HcSr04DistanceSensorDriver(HcSr04DistanceSensorDriverSettings settings, ILogger<HcSr04DistanceSensorDriver> logger)
+        {
+            Id = settings.SensorId;
+            Angle = settings.Angle;
+            Sensor = new Hcsr04(settings.TriggerPin, settings.EchoPin, PinNumberingScheme.Logical);
+            MeasuringCancellationToken = new CancellationTokenSource();
+            Task.Run(MeasurementCycle);
+            Settings = settings;
+            Logger = logger;
+        }
+
+        private async void MeasurementCycle()
+        {
+            var token = MeasuringCancellationToken.Token;
+            while (!token.IsCancellationRequested)
+            {
+                await Task.Delay(Settings.MeasuringInterval);
+                try
+                {
+                    if (SonarDistanceChanged != null)
+                    {
+                        var measurement = Sensor.Distance / 100; // Convert from cm to m
```

Comment:
If we use meters here, we should use meters everywere. For current robots, meters seems to big unit, but makes sense to keep it for consitency. I like keeping meters, but lets let everyone be aware of this

Reply:
Thanks for the question, I wanted to share my answer: it's reasonable to use a single set of units all over our software (so if we see distance here and there, we know that it's in both cases measured in the same units).
Considering this, let's use SI  ([link to the wikipedia](https://en.wikipedia.org/wiki/International_System_of_Units)).
And as for angle, it's easier to use radians because all the trigonometric functions accept radians as arguments. This way we may avoid unnecessary transformations.

Reply:
Probably, we should put these considerations in a file like "CONTRIBUTE.md". What do you think?

Reply:
Sure! feel free to do so